libvast/vast/data.hpp

```diff
@@ -318,23 +313,26 @@ bool evaluate(const data& lhs, relational_operator op, const data& rhs);
 
 // -- convertible -------------------------------------------------------------
 
-bool convert(const list& xs, json& j);
-bool convert(const map& xs, json& j);
-bool convert(const record& xs, json& j);
-bool convert(const data& xs, json& j);
+template <class T, class... Opts>
+data to_data(const T& x, Opts&&... opts) {
+  data d;
+  if (convert(x, d, std::forward<Opts>(opts)...))
+    return d;
+  return {};
+}
 
 /// Converts data with a type to "zipped" JSON, i.e., the JSON object for
 /// records contains the field names from the type corresponding to the given
 /// data.
-bool convert(const data& x, json& j, const type& t);
+// bool convert(const data& x, json& j, const type& t);
 
 caf::error convert(const record& xs, caf::dictionary<caf::config_value>& ys);
```

Comment:
Those 3 convert functions are quite asymmetric to many other convert function in vast, which return bool to manifest the result of conversion. while `if (convert(x,y))` is test on success for return-bool-based convert functions it is a test on fail for return-caf::error-based convert function.

Reply:
Yes, they were overlooked in a past refactoring. Generally, `convert` overloads must return something that is explicitly convertible to bool, but just returning bool is preferred since the errors aren't checked anyways.